docs/tools/list.md

```diff
@@ -0,0 +1,51 @@
+# ADK Tools list
+
+Looking for a pre-built [ADK Tool](/adk-docs/tools/) to use in an ADK agent 
+workflow? Here is a list of available ADK Tools:
+
+### ADK-provided tools
+
+*   **[Google Search](/adk-docs/tools/built-in-tools/#google-search)**:
+    Perform web searches using Google Search.
+*   **[Code Execution](/adk-docs/tools/built-in-tools/#code-execution)**:
+    Execute code using Gemini 2 models.
+
+### Google Cloud tools
+
+*   **[GKE Code Executor](/adk-docs/tools/built-in-tools/#gke-code-executor)**:
+    Run AI-generated code in a secure and scalable GKE Sandbox environment.
+*   **[Vertex AI Search](/adk-docs/tools/built-in-tools/#vertex-ai-search)**:
+    Search across your private, configured data stores in Vertex AI Search.
+*   **[Vertex AI RAG Engine](/adk-docs/tools/built-in-tools/#vertex-ai-rag-engine)**:
+    Perform private data retrieval using Vertex AI RAG Engine.
+*   **[BigQuery Tools](/adk-docs/tools/built-in-tools/#bigquery)**:
+    Connect with BigQuery to retrieve data and perform analysis.
+*   **[Spanner Tools](/adk-docs/tools/built-in-tools/#spanner)**:
+    Interact with Spanner to retrieve data, search, and execute SQL.
+*   **[Bigtable Tools](/adk-docs/tools/built-in-tools/#bigtable)**:
+    Interact with Bigtable to retrieve data and and execute SQL.
+*   **[MCP Toolbox for Databases](/adk-docs/tools/google-cloud-tools/#toolbox-tools-for-databases)**:
+    Connect a wide variety of databases to your agents.
+*   **[Apigee API Hub Tools](/adk-docs/tools/google-cloud-tools/#apigee-api-hub-tools)**:
+    Turn any documented API from Apigee API hub into a tool.
+*   **[Application Integration Tools](/adk-docs/tools/google-cloud-tools/#application-integration-tools)**:
+    Link your agents to enterprise applications using Integration Connectors.
+
+### Third-party tools
+
+*   **[LangChain Tools](/adk-docs/tools/third-party-tools/#using-langchain-tools)**:
+    Integrate tools from the LangChain ecosystem.
+*   **[CrewAI tools](/adk-docs/tools/third-party-tools/#using-crewai-tools)**:
+    Integrate tools from the CrewAI library.
+*   **[OpenAPI Integration](/adk-docs/tools/openapi-tools/)**:
+    Generate callable tools directly from an OpenAPI Specification.
+
+## Build your tools
+
+If these tools do not meet your needs, you can build tools for your ADK
+workflows using the following guides:
+
+*   **[Function Tools](/adk-docs/tools/function-tools/)**: Build custom tools for
```

Comment:
: write any python code and equip your agent with that code as a tool.

Reply:
this suggested language is a little too complicated (also not just Python: Java as well) These descriptions need to be scannable. "write any python code" sounds like you are empowering the Agent to build python code (not actually what this does).